packages/subgraph/src/utils.ts

```diff
@@ -109,34 +110,34 @@ export function getIsListedToken(
     token: Token,
     resolverAddress: Address
 ): Token {
-    let resolverContract = Resolver.bind(resolverAddress);
+    const resolverContract = Resolver.bind(resolverAddress);
     const RESOLVER_ADDRESS = getResolverAddress();
-    let version = resolverAddress.equals(RESOLVER_ADDRESS) ? "test" : "v1";
-    let result = resolverContract.try_get(
+    const version = resolverAddress.equals(RESOLVER_ADDRESS) ? "test" : "v1";
+    const result = resolverContract.try_get(
         "supertokens." + version + "." + token.symbol
     );
-    let superTokenAddress = result.reverted ? ZERO_ADDRESS : result.value;
+    const superTokenAddress = result.reverted ? ZERO_ADDRESS : result.value;
     token.isListed = token.id == superTokenAddress.toHex();
 
     return token;
 }
 
-export function updateTotalSupplyForNativeSuperToken(
-    token: Token,
+/**
+ * Gets and sets the total supply for TokenStatistic of a SuperToken upon initial creation
+ * @param tokenStatistic 
+ * @param tokenAddress 
+ * @returns TokenStatistic
+ */
+export function getInitialTotalSupplyForSuperToken(
```

Comment:
logic change alert: change function to be called for all super tokens (not just custom)

Reply:
I wonder why this changed?

Reply:
it didn't make sense to only execute this function for native asset/custom super tokens